numpy/testing/_private/utils.py

```diff
@@ -695,17 +695,23 @@ def isnumber(x):
     def istime(x):
         return x.dtype.char in "Mm"
 
-    def chk_same_position(x_id, y_id, hasval='nan'):
-        """Handling nan/inf: check that x and y have the nan/inf at the same
-        locations."""
-        try:
-            assert_array_equal(x_id, y_id)
-        except AssertionError:
+    def func_assert_same_pos(x, y, func=isnan, hasval='nan'):
+        """Handling nan/inf: combine results of running func on x and y,
+        checking that they are True at the same locations."""
+        # Both the cast to bool_ and the != True comparison are done
```

Comment:
Can you move the cast to bool to the scalar case below?

Reply:
I needed to cast both `x_id` and `y_id`, but that can indeed be done at the end. That does look a bit nicer, so done.